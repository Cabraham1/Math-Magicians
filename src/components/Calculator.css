:root {
  --primary: rgb(153, 152, 152);
  --digit: #f5f5f5;
  --operator: rgb(255, 144, 104);
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.lets {
  margin-left: 2rem;
}

.global {
  width: 100%;
  max-width: 400px;
  margin: 0 auto;
  background-color: var(--primary);
}

.answer-span {
  background-color: var(--primary);
  text-align: right;
  padding: 1.5rem;
  color: var(--digit);
}

.box-one,
.box-two,
.box-three,
.box-four,
.box-five {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: none;
  outline: none;
  font-size: 1rem;
  appearance: none;
  margin-right: 1rem;
  background: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.box-one button,
.box-two button,
.box-three button,
.box-four button {
  margin: 1px 1px 1px 0;
  padding: 0.5rem;
}

.zero {
  grid-column: 1 / span 2;
}

.box-five button {
  margin: 1px 1px 1px 0;
  padding: 0.5rem;
}

.operator,
.number,
.empty {
  border-width: 0;
  font-weight: bold;
  font-size: 25px;
  cursor: pointer;
}

.empty {
  background-color: var(--digit);
}

.number {
  background-color: var(--digit);
}

.operator {
  background-color: var(--operator);
  font-size: 30px;
}
