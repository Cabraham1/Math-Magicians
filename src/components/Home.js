import React from "react";

const Home = () => (
  <div className="home-container">
    <h1>Welcome to our page</h1>
    <p>
      Mathematics is the science that deals with the logic of shape, quantity
      and arrangement. Mathematics plays a vital role in all aspects of life,
      whether in everyday matters such as time tracking, driving, cooking, or
      jobs such as accounting, finance, banking, engineering, and software.
    </p>

    <p>
      Mathematics is the cradle of all creations, without which the world cannot
      move an inch. It makes our life orderly and prevents chaos. Certain
      qualities that are nurtured by mathematics are power of reasoning,
      creativity, abstact or spatial thinking, critical thinking and even
      effective communication.
    </p>
  </div>
);

export default Home;
