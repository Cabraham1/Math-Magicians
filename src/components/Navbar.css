.navbar {
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 1rem;
  font-weight: bold;
  color: #f2f2f2;
}

.navbar a {
  margin-left: 16px;
  text-decoration: none;
  padding: 6px;
  color: #f2f2f2;
  list-style: none;
}

.navbar a:hover {
  color: rgb(255, 144, 104);
}

.nav-items {
  display: flex;
  justify-content: space-between;
}

.nav-items li {
  list-style: none;
}

.navbar .active {
  font-weight: 600;
  border-bottom: 2px solid var(--orange);
}