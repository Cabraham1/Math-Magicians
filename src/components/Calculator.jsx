import React from 'react';
import './Calculator.css';

function Calculator() {
  return (
    <div>
      <div className="global">
        <div className="answer-span">
          <span>0</span>
        </div>
        <div className="box-one">
          <button type="button" className="btn delete number">
            AC
          </button>
          <button type="button" className="btn  number number">
            +/-
          </button>
          <button type="button" className="btn clear number">
            %
          </button>
          <button type="button" className="btn  equal-to operator">
            รท
          </button>
        </div>
        
        </div>
      </div>
    </div>
  );
}

export default Calculator;
