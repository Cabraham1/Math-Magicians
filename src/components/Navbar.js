import { NavLink } from 'react-router-dom';
import './Navbar.css';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },

    {
      id: 3,
      path: '/quotes',
      text: 'Quotes',
    },
  ];

  return (
    <div className="navbar">
      <h2>Math Magician</h2>
      <ul className="nav-items">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink
              to={link.path}
              style={({ isActive }) => ({
                color: isActive ? 'orangered' : 'white',
                textDecoration: isActive ? 'underline' : null,
              })}
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Navbar;
