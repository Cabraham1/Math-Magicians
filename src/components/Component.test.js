import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Home from './Home';
import Quote from './Quotes';
import Calculator from './Calculator';
import Navbar from './Navbar';

it('renders Home correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Quote correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Calculator correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Navbar correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
