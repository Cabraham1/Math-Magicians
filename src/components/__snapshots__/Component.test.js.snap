// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<div
  className="main-container"
>
  <h2
    className="lets"
  >
    Lets do some Math
  </h2>
  <div
    className="global"
  >
    <div
      className="answer-span"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="box-one"
    >
      <button
        className="btn delete number"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn  number number"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn clear number"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn  equal-to operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="box-two"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="box-three"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn number  operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="box-four"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="box-five"
    >
      <button
        className="btn number zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn number dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders Home correctly 1`] = `
<div
  className="home-container"
>
  <h1>
    Welcome to our page
  </h1>
  <p>
    Mathematics is the science that deals with the logic of shape, quantity and arrangement. Mathematics plays a vital role in all aspects of life, whether in everyday matters such as time tracking, driving, cooking, or jobs such as accounting, finance, banking, engineering, and software.
  </p>
  <p>
    Mathematics is the cradle of all creations, without which the world cannot move an inch. It makes our life orderly and prevents chaos. Certain qualities that are nurtured by mathematics are power of reasoning, creativity, abstact or spatial thinking, critical thinking and even effective communication.
  </p>
</div>
`;

exports[`renders Navbar correctly 1`] = `
<div
  className="navbar"
>
  <h2>
    Math Magician
  </h2>
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "orangered",
            "textDecoration": "underline",
          }
        }
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "textDecoration": null,
          }
        }
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quotes"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "textDecoration": null,
          }
        }
      >
        Quotes
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders Quote correctly 1`] = `
<div>
  Something went wrong!
</div>
`;
